internal/pkg/rpaas/manager.go

```diff
@@ -163,3 +166,108 @@ type CertificateData struct {
+	Replicas    *int32                               `json:"replicas,omitempty"`
+	Plan        string                               `json:"plan,omitempty"`
+	Locations   []v1alpha1.Location                  `json:"locations,omitempty"`
+	Service     *nginxv1alpha1.NginxService          `json:"service,omitempty"`
+	Autoscale   *v1alpha1.RpaasInstanceAutoscaleSpec `json:"autoscale,omitempty"`
+	Binds       []v1alpha1.Bind                      `json:"binds,omitempty"`
+	Team        string                               `json:"team,omitempty"`
+	Name        string                               `json:"name,omitempty"`
+	Description string                               `json:"description,omitempty"`
+	Tags        []string                             `json:"tags,omitempty"`
+}
+
+func NewInfoInstance(instance *v1alpha1.RpaasInstance) *InfoBuilder {
+	info := &InfoBuilder{
+		Replicas:  instance.Spec.Replicas,
+		Plan:      instance.Spec.PlanName,
+		Locations: instance.Spec.Locations,
+		Service:   instance.Spec.Service,
+		Autoscale: instance.Spec.Autoscale,
+		Binds:     instance.Spec.Binds,
+		Name:      instance.ObjectMeta.Name,
+	}
+
+	if desc, ok := instance.ObjectMeta.Annotations["description"]; ok {
+		info.Description = desc
+	}
+	info.parseTags(instance)
+
+	return info
+}
+
+func (i *InfoBuilder) SetAddress(ctx context.Context, manager RpaasManager, instanceName string) error {
+	address, err := manager.GetInstanceAddress(ctx, instanceName)
+	if err != nil {
+		return err
+	}
+	i.Address = &InstanceAddress{
+		Ip: address,
+	}
+	return nil
+}
+
+func (i *InfoBuilder) parseTags(instance *v1alpha1.RpaasInstance) {
+	re1 := regexp.MustCompile(`.*?,`)
+	re2 := regexp.MustCompile(`,\w*$`)
+	tags := re1.FindAll([]byte(instance.ObjectMeta.Annotations["tags"]), -1)
+	last_tag := re2.Find([]byte(instance.ObjectMeta.Annotations["tags"]))
+
+	var new_tags []string
+	for _, tag := range tags {
+		new_tags = append(new_tags, strings.TrimSuffix(string(tag), ","))
+	}
+	if last_tag != nil {
+		new_tags = append(new_tags, strings.TrimSuffix(string(last_tag), ","))
+	}
+
+	if len(new_tags) > 0 {
+		i.Tags = new_tags
+	}
+}
+
+func (i *InfoBuilder) SetTeam(instance *v1alpha1.RpaasInstance) error {
+	for key, _ := range instance.ObjectMeta.Annotations {
+		matched, err := regexp.Match(`team-owner`, []byte(key))
+		if err != nil {
+			return err
+		}
+		if matched {
+			i.Team = instance.ObjectMeta.Annotations[key]
+			return nil
+		}
+	}
+
+	for key, _ := range instance.Labels {
+		matched, err := regexp.Match(`team-owner`, []byte(key))
+		if err != nil {
+			return err
+		}
+		if matched {
+			i.Team = instance.Labels[key]
+			return nil
+		}
+	}
+
+	for key, _ := range instance.Spec.PodTemplate.Labels {
+		matched, err := regexp.Match(`team-owner`, []byte(key))
+		if err != nil {
+			return err
+		}
+		if matched {
+			i.Team = instance.Spec.PodTemplate.Labels[key]
+			return nil
+		}
+	}
+
+	return errors.New("instance has no team owner")
+}
```

Comment:
This file shouldn't have "business" logic else struct and interface definitions. It would be great to move those functions in favor a new method within RPaaS manager, as shown below.

```golang
GetInstanceInfo(ctx context.Context, instance string) (InfoInstance, error)
```

This new method should gathering all instance's resources (`RpaasInstance`, `RpaasFlavor`, `RpaasPlan`, `Nginx`, etc) and assemble those into a new `InfoInstance` struct (looks like yours `InfoBuilder`).